src/mongo/delegates/Updates.class.php

```diff
@@ -659,6 +668,75 @@ protected function applyChangeSet(\Tripod\ChangeSet $cs, $originalCBDs, $context
         }
     }
 
+    private function getAdditionsRemovalsGroupedByNsPredicate($cs,$additions,$removals)
+    {
+        $additionsGroupedByNsPredicate = $this->getChangesGroupedByNsPredicate($cs,$additions);
+        $removalsGroupedByNsPredicate = $this->getChangesGroupedByNsPredicate($cs,$removals);
+
+        $mergedResult = array();
+        foreach($additionsGroupedByNsPredicate as $predicate => $values)
+        {
+            if (!isset($mergedResult[$predicate]))
```

Comment:
I don't think this is necessary -  you can assume that it's not set at this point.  It is necessary for the removals, however.